### ISSUE TITLE
Add `--git` option to `mtenv install` to install version by git identifier

### ISSUE THREAD

**Issue by Jens0512** [myst-lang/mtenv#3]
ATM, `mtenv` has two ways to install a myst version:
```
$ mtenv install <github-release tag>
```
or
```
$ mtenv install <git commit SHA>
```
Where `<github-release tag>` is e.g. `v0.6.1` ([myst/releases/tag/v0.6.1](https://github.com/myst-lang/myst/releases/tag/v0.6.1)), and maybe
`<git commit SHA>`: `32c2a6896ea2e1897ec519853844e30b58326ee1` (or shortened: `32c2a68`) ([myst#199](https://github.com/myst-lang/myst/commit/32c2a6896ea2e1897ec519853844e30b58326ee1)).

While this works fine, if you want to test with a nightly myst version, you have to manually find a *commit SHA* from github to install the version. The convention regarding nightly git versions, is to use the `master` branch or a `nightly`  tag, neither of which are currently supported by `mtenv`.

## Proposal
Initially proposed in the Discord was (quoting myself) "something à la `mtenv install master`", to which @faultyserver suggested an excellent improvement:
```
$ mtenv install --git master
```
Where `--git` takes an argument: a git branch, tag or SHA. Rendering github release tags the only way to install myst versions without the `--git` option, which is good, as you only get a non-release version if you specifically say so.

**Comment by faultyserver** [comment]
Is this handled by #4 as well?

**Comment by Jens0512** [comment]
No, I thought it was best to put my foot down, and just release what I had before it added to the chaos...
